.github/workflows/golint.yml

```diff
@@ -0,0 +1,49 @@
+name: golangci-lint
+on:
+  push:
+    tags:
+      - v*
+    branches:
+      - master
+      - main
```

Comment:
while we don't have a `main` branch, perhaps we should have an item to migrate to this naming strategy

Reply:
I have no opinion on this :) , either is fine with me 
It's in here because I shamelessly stole the config file from the official golangci-lint github action and did not notice it